pkg/networkservice/common/crossnse/server.go

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package crossnse provides a NetworkServiceServer chain element that tracks local Cross connect Endpoints and call them first
+// their unix file socket as the clienturl.ClientURL(ctx) used to connect to them.
+package crossnse
```

Comment:
Hmm... I wonder if we can come up with a better name here for the package.  Functionally... what we are doing is interposing xconn NSEs... 

perhaps something like 

'interpose'

if generic?  Totally open to discussion here.  Good naming will make things much easier for subsequent consumers.

Reply:
interpose looks good for me :-)